### ISSUE TITLE
Guided Tutorial

### ISSUE THREAD

**Issue by therationalpi** [Monika-After-Story/DDLCModTemplate#10]
A guided tutorial that steps the player through the steps of writing a simple scene with renpy and building a distribution needs to be added for version 0.2.0

**Comment by haloff1** [comment]
We discussed this issue in the past issue "More polished example script". For reference, here is what I proposed:

....................................................................

The idea is to make Monika teaches the modder how to make a monika route mod while explaining each step. For example at the beginning, she will asks the player...:

Tutorial PART 1

    Copy-paste the DDLC Mod Template Directory in Ren'Py's directory and then go inside the new folder
    Make an empty text file called "monika_route_script.rpy"
    Edit script.rpy and replace the line "call example_chapter from _call_example_chapter" by "call monika_route from _call_monika_route"
    Edit monika_route_script.rpy and add "label monika_route:"
    3.1 Explain the meaning of call and label

TUTORIAL PART 2 MUSIC AND SOUND

    Add " stop music fadeout 2.0"
    Add "play music t2"
    5.1 Explain the meaning of each keyword and how music works in Ren'Py
    5.2 Show the modder where they can find the meaning of t2 (definition.rpy)
    5.3 Invite the modder to see what happens when they write "music t3"

TUTORIAL PART 3 BG AND TRANSITION

    Add " scene bg residential_day"
    Add " with dissolve_scene_full"
    7.1 Explain the meaning of the two lines
    7.2 Show several other bg used in DDLC

TUTORIAL PART 4 Dialogue

    Add " "It has been four days since I joined the Literature Club. Today is Saturday and I finally decided to confess my feelings to Monika""
    8.1 Explain how dialogue works

TUTORIAL PART 5 CHOICE AND VARIABLE

    Add "menu".
    And so on. Monika will basically teach the modder how to make something like The Question.

I think this kind of tutorial is much better at teaching a new modder than a tutorial where all you do is reading.

....................................................................

I'm currently working on it. I will inform you my progression today at 23:00 UTC+01.

**Comment by haloff1** [comment]
Here is the script that Monika would ask the modder to write. Each time she asks the modder to write a new keyword, she would give explanation about it.  The writing of the script would be done in several steps. Each step corresponds to a tutorial.

................................

label monika_route:

    stop music fadeout 2.0
    play music t2
    
    scene bg residential_day
    with dissolve_scene_full
    
    mc "It has been four days since I joined the Literature Club. Today is Saturday and I finally decided to confess my feeling to Monika."
    
    menu:
        
        mc "I told her to meet me..."
        "At the literature club room":
                $ meeting_place = "club_room"
        "In front of my house":
                $ meeting_place = "my_house"
                
    mc "I can't wait to meet her!"
    
    if meeting_place == "club_room":
        scene bg club_day
        with wipeleft_scene
        
    elif meeting_place == "my_house":
        scene bg house
        with wipeleft_scene
        
    stop music fadeout 2.0
    play music t2
        
    mc "She is already waiting for me when I arrive."
    
    show monika 1b at t11 zorder 2
    
    m "Hi [player]!"
    
    mc "You're already here? I hope I didn't make you wait for too long."
    
    m 2a "Don't worry, it's me who's early."
    
    show monika 5a at f11 zorder 3
    
    m "So what did you want to tell me?"
    
    menu:
         
        mc "Right. Monika, ..."
        "I love you! Please go out with me!":
            jump monika_normal_ending
        "You are everything to me! Please marry me!":
            jump monika_good_ending
    
label monika_normal_ending:
    
    scene dark
    with dissolve_scene_full 
    
    mc "She accepted my confession and we became lovers."
    
    "NORMAL ENDING."
    
    return
    
label monika_good_ending:
    
    scene white
    with dissolve_scene_full 
    
    mc "She gladly accepted my proposal and we got married one year later."
    
    "GOOD ENDING."
    
    return

..............................

There would be a cheat script where the script is already written to help the modder if they're stuck.

**Comment by haloff1** [comment]
I plan to divide the tutorials in the following way:

**Tutorial 1**: “ How To Make A Simple Route, Part 1” 

From : ... (copy game folder, use ren'py bla bla)
To : label monika_route:

**Tutorial 2:** “Route Part 2, Music”

From : label monika_route:
To : play music t2

**Tutorial 3:** “Route Part 3, Scene”

From : play music t2
To : with dissolve_scene_full

**Tutorial 4:** “Route Part 4, Dialogue”

From : mc "It has been four days since I joined the Literature Club. Today is Saturday and I finally decided to confess my feeling to Monika."
To : mc "It has been four days since I joined the Literature Club. Today is Saturday and I finally decided to confess my feeling to Monika."

**Tutorial 5:** “Route Part 5, Menu”

From : menu:
To : mc "I can't wait to meet her!"

**Tutorial 6 :** « Route Part 6, Logic Statement”

From : mc "I can't wait to meet her!"
To : with wipeleft_scene

**Tutorial 7:** “Route Part 7, Sprite”

From : with wipeleft_scene
To : show monika 1b at t11 zorder 2

**Tutorial 8:** “Route Part 8, Position”

From : show monika 1b at t11 zorder 2
To : show monika 5a at f11 zorder 3

**Tutorial 9:** “Route Part 9, Ending”

From : menu:
To : end of the script

**Tutorial 10.** "How To Build The Mod"

**Comment by haloff1** [comment]
Update.
Sigh. I didn't do nearly as much as I wanted to...
I finished the first tutorial but expressions and poses are missing. Monika's dialogues should be checked too.

I plan to finish Tutorial 2,3,4,5 tomorrow evening and Tutorial 6,7,8,9 Tuesday evening. 

You can find what I did until now here:
https://github.com/haloff1/DDLCModTemplate/blob/master/game/tutorials.rpy

**Comment by haloff1** [comment]
Update.
I finished the second tutorial but I'm far behind schedule... Like last time, I added it to tutorial.rpy in my repository.
I will probably not finish them until Friday...

**Comment by haloff1** [comment]
Update.
Tutorial 3 done.
Tutorial 4 done.
Yeah, it's going to take me the whole week.

**Comment by haloff1** [comment]
Update.
Tutorial 5 done.
Tutorial 6 done.
Tutorial 7 done.
Tutorial 8 done.
Tutorial 9 done.

Everything is finished!

Or not. I wrote the tutorials but we still need to check the spelling, grammar and whether they fit Monika's personality (spoiler: they don't). We also need to add expressions.

I would like your help for that. 

At the end of the last tutorial, a persistent variable is created :  persistent.DDLC_mod_template_tutorial_finished . When a player plays Monika After Story, I want the game to import the persistent file of DDLC Mod Template and check if that variable is True or None. If True, a new scene should be unlocked.

Please download the  latest version of my tutorials here https://github.com/haloff1/DDLCModTemplate .
You will understand what I mean.

EDIT : To see the tutorials, you first need to finish reading the introduction done by Monika (about Ren'Py, how much you know about programming and so on).

**Comment by therationalpi** [comment]
I think a better way to do this is to use the persistent.playthrough variable that's already in the game.

**Comment by haloff1** [comment]
persistent.playthrough is a variable already used by the original game. 
If we use that variable to see whether the player already cleared the tutorials of DDLC Mod Template then we will lose the information of what was the latest Act the player played. 

If they're some special scene in MAF that depends on the current Act in DDLC, then we won't be able to unlock them if we already used persistent.playthrough for the tutorials. 

About the tutorials, I'm currently reviewing them. I checked and added expressions to the first one. I think I may be able to do the rest today.

EDIT : Added expressions to tutorial 1,2,3,4,5,6,7.
